hydromt/metadata_resolvers/convention_resolver.py

```diff
@@ -0,0 +1,109 @@
+"""MetaDataResolver using HydroMT naming conventions."""
+from itertools import product
+from string import Formatter
+from typing import TYPE_CHECKING, Any
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+
+from hydromt.nodata import NoDataStrategy
+from hydromt.typing import Bbox, TimeRange
+
+from .metadata_resolver import MetaDataResolver
+
+if TYPE_CHECKING:
+    from hydromt.data_sources.data_source import DataSource
```

Comment:
did not know this construct was a thing, TIL.

Reply:
You will have to then type hint using `"`, like `def f(a: "MyType"):`